### ISSUE TITLE
AutoResizer dom exception and react-lite

### ISSUE THREAD

**Issue by drcmda** [bvaughn/react-virtualized#385]
I get this when switching my build to production:

    Uncaught TypeError: Cannot read property '__resizeTriggers__' of null(…)

It relies on a parentNode in a ref hook, can't find it and complains down the chain. This happens in react-lite.

**Comment by bvaughn** [comment]
Hey @drcmda,

Specific problems with applications may be better directed to Stack Overflow or Gitter. You may want to try there before filing a Github issue.

If you don't have any luck with those channels, or if you strongly prefer filing a Github issue- read the new-issue guidelines (the text in the issue box when you first open it) which explains that a repro case is required. (How can I possible help fix your production app, knowing nothing about it?) You can provide repro steps by sharing a Plunkr or a pointer to a Github project I can checkout and run.

I'm going to close this issue in the meanwhile as it lacks sufficient information to be actionable.

**Comment by bvaughn** [comment]
By the way- no need to open a new issue. If you're able to provide repro information, just add it to this issue and I can re-open it.

**Comment by bvaughn** [comment]
FYI I've created a reproduction case [here](https://plnkr.co/edit/igZ6BmC6z8P1N8PSyaPg). A little bit of massaging is necessary so that `react-light` mirrors `react-with-addons` and `react-dom`:
```js
React.addons = {
  shallowCompare: React.PureComponent.prototype.shouldComponentUpdate
};
ReactDOM = React;
```
...but once that's done, I see the error you've mentioned.

Given that the exact same JSX works fine with `react` (see [here](https://plnkr.co/edit/hrY4cQdbN9Y1ShDhcSRo?p=preview)) then I think this has something to do with a difference between `react-lite` and React. I've never used `react-lite` and I don't really have the time to dig into trying to support it so- for the time being- this issue is going to remain closed.

If you'd like to dig in and submit a PR though- it would be welcome! 😄

**Comment by bvaughn** [comment]
Btw the problem can be hacked around somewhat by storing the ref to `AutoSizer` instead and waiting to access `parentNode` until `componentDidMount`:
```diff
diff --git a/source/AutoSizer/AutoSizer.js b/source/AutoSizer/AutoSizer.js
index 1c6862e..9592c16 100644
--- a/source/AutoSizer/AutoSizer.js
+++ b/source/AutoSizer/AutoSizer.js
@@ -44,6 +44,9 @@ export default class AutoSizer extends Component {
   }
 
   componentDidMount () {
+    // Delay access of parentNode for react-lite
+    this._parentNode = this._autoSizer.parentNode
+
     // Defer requiring resize handler in order to support server-side rendering.
     // See issue #41
     this._detectElementResize = require('../vendor/detectElementResize')
@@ -122,6 +125,6 @@ export default class AutoSizer extends Component {
 
   _setRef (autoSizer) {
     // In case the component has been unmounted
-    this._parentNode = autoSizer && autoSizer.parentNode
+    this._autoSizer = autoSizer
   }
 }
```

However, this just leads to another error inside of `react-lite`'s implementation of `shallowCompare`:
> Uncaught TypeError: Cannot read property 'props' of undefined

**Comment by bvaughn** [comment]
Ooh, looks like the `shallowCompare` error could be resolved by mapping args:
```js
React.addons = {
  shallowCompare: function(context, nextProps, nextState) {
    return React.PureComponent.prototype.shouldComponentUpdate(nextProps, nextState);
  }
};
ReactDOM = React;
```

**Comment by bvaughn** [comment]
Looks like some stuff still doesn't work quite the same, which causes problems with other HOCs like `CellMeasurer`. Given this initial impression, I'm not sure I really want to go down the rabbit hold of claiming to support `react-lite`.

**Comment by bvaughn** [comment]
I've just released version 7.24.1 with `AutoSizer` support for `react-lite`. I'm able to run, for example, the [hover demo](https://github.com/bvaughn/react-virtualized/tree/master/playground/hover.html) by pointing at the UMD build of `react-lite` and including the above `React` + `ReactDOM` initialization lines.

Beyond this, I don't _officially_ support `react-lite` and so some HOCs (eg `CellMeasurer`) may not work. Cheers.

**Comment by drcmda** [comment]
You're amazing, thanks and cheers!
